File: WebApi/api/api/Models/CatalogColor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace api.Models
{
    public class CATALOG_COLOR
    {
        [Key]
        public long catalog_color_id { get; set; }
        public long catalog_id { get; set; }

        [StringLength(10)]
        public string pdcolor_code { get; set; }

        [StringLength(200)]
        public string pic_file_path { get; set; }

        [StringLength(int.MaxValue)]
        public string pic_base64 { get; set; }
        [StringLength(15)]
        public string created_by { get; set; }

        public DateTime created_at { get; set; }
        [StringLength(15)]
        public string updated_by { get; set; }

        public DateTime updated_at { get; set; }
    }
}

File: web/src/app/_model/catalog-color.ts
import { AppSetting } from '../_constants/app-setting';

export class CatalogColorView 
{
    public catalog_color_id: number = undefined;
    public catalog_id: number = undefined;
    public pddsgn_name:string = ""
    public pdcolor_code: string = "";
    public pdcolor_name: string = "";
    public pic_file_path: string = "";
    public pic_base64: string = "";
    public created_by: string = "";
    public created_at: any = null;
    public updated_by: string = "";
    public updated_at :any= null;

}

export class CatalogColorSearchView
{
    public pageIndex: number = 1;
    public itemPerPage: number = AppSetting.itemPerPage;
    public catalog_id: number = undefined;
    
}